helm/cluster-autoscaler-app/templates/deployment.yaml

```diff
@@ -24,7 +24,11 @@ spec:
         operator: "Exists"
         key: node-role.kubernetes.io/master
       nodeSelector:
-{{- if semverCompare ">=1.24.0" .Capabilities.KubeVersion.Version }}
+{{- if ((.Values.node).nodeSelector) }}
+        {{- with .Values.node.nodeSelector }}
+{{ toYaml . | indent 8 }}
+        {{- end }}
+{{- else if semverCompare ">=1.24.0" .Capabilities.KubeVersion.Version }}
```

Comment:
```suggestion
        {{- if .Values.node.nodeSelector }}
        {{- toYaml .Values.node.nodeSelector | nindent 8 }}
        {{- end }}
        {{- else if semverCompare ">=1.24.0" .Capabilities.KubeVersion.Version }}
```

Reply:
I can fix the indentation, but removing the parentheses causes issues if .Values.node is not defined, because of helm/go templating:
https://github.com/helm/helm/issues/8026
https://stackoverflow.com/questions/59795596/how-to-make-nested-variables-optional-in-helm/68807258#68807258

Reply:
```suggestion
        {{- if ((.Values.node).nodeSelector) }}
        {{- toYaml .Values.node.nodeSelector | nindent 8 }}
        {{- else if semverCompare ">=1.24.0" .Capabilities.KubeVersion.Version }}
```

Reply:
Added in last commit, ty!